pandas/core/reshape/concat.py

```diff
@@ -500,7 +500,9 @@ def _get_concat_axis(self):
                 else:
                     return ibase.default_index(len(self.objs))
             else:
-                return ensure_index(self.keys)
+                new_axis = ensure_index(self.keys)
```

Comment:
use .set_names as this could modify self.keys

Reply:
done